src/main/kotlin/uk/gov/justice/digital/hmpps/hmppssubjectaccessrequestworker/services/WebClientService.kt

```diff
@@ -0,0 +1,33 @@
+package uk.gov.justice.digital.hmpps.hmppssubjectaccessrequestworker.services
+
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.http.HttpStatusCode
+import org.springframework.stereotype.Service
+import org.springframework.web.reactive.function.client.WebClient
+import uk.gov.justice.digital.hmpps.hmppssubjectaccessrequestworker.gateways.HmppsAuthGateway
+import uk.gov.justice.digital.hmpps.hmppssubjectaccessrequestworker.models.SubjectAccessRequest
+
+@Service
+class WebClientService(@Autowired val hmppsAuthGateway: HmppsAuthGateway) {
+
+  fun getClient(url: String): WebClient {
+    return WebClient.create(url)
+  }
+  fun getToken(): String {
+    return hmppsAuthGateway.getClientToken()
+  }
+
+  fun getUnclaimedSars(client: WebClient, token: String): Array<SubjectAccessRequest>? {
```

Comment:
Pls can we make the names uniform? I.e., 'getUnclaimedSar', 'claimSar' and 'completeSar', or 'getUnclaimed', 'claim', and 'complete'?

Also does this function return multiple Sars or 1 Sar - can't remember - but if using the Sar variation above, can we check whether it should be Sars or Sar?